src/util/fetch.js

```diff
@@ -0,0 +1,59 @@
+import Observer from './observer';
+
+/**
+ * Perform an fetch request
+ *
+ * @param {Options} options Description
+ * dofetch({
+ *   url: "https://www.waveform.al/sample.json"
+ * })
+ *
+ * @returns {Object} Observer instance
+ */
+export default function dofetch(options) {
+    if (self.fetch) {
+        const instance = new Observer();
+        const fetchRequest = new Request(options.url);
+        let fetchHeaders = new Headers();
+
+        // check if headers and credentials are added
+        if (options) {
+            if (options.requestHeaders) {
+                // add custom request headers
+                options.requestHeaders.forEach(header => {
+                    fetchHeaders.append(header.key, header.value);
+                });
+            }
+        }
+
+        // set default fetch options
+        const fetchOptions = {
+            method: (options.method || "GET"),
+            headers: fetchHeaders,
+            mode: (options.mode || "cors"),
+            credentials: (options.credentials || "same-origin"),
+            cache: (options.cache || "default")
+        };
+
+        // do the fetch
+        fetch(fetchRequest,fetchOptions).then(function(response) {
+          if (response.ok) {
+            return response.blob();
+          } else{
+            instance.fireEvent('HTTP error status ', response.status)
```

Comment:
I think here you might need to throw an exception? Otherwise the next `then` will try to run.

Reply:
I can add it, but currently it will return the `instance.fireEvent` event as it catches the error and it won't go to the next `then`. The error message will say `Failed to fetch` by default if the url cannot be resolved.

Reply:
If you don't throw an error in this `then`, it will try to run the next `then` (which will fail because the `blobResponse` will be undefined).

Reply:
fixed.